lib/DXIL/DxilModule.cpp

```diff
@@ -288,11 +288,39 @@ unsigned DxilModule::GetGlobalFlags() const {
   return Flags;
 }
 
+static bool RequiresRaytracingTier1_1(const DxilSubobjects *pSubobjects) {
+  if (!pSubobjects)
+    return false;
+  for (const auto &it : pSubobjects->GetSubobjects()) {
+    switch (it.second->GetKind()) {
+    case DXIL::SubobjectKind::RaytracingPipelineConfig1:
+      return true;
+    case DXIL::SubobjectKind::StateObjectConfig: {
+      uint32_t configFlags;
+      if (it.second->GetStateObjectConfig(configFlags) &&
+          ((configFlags &
+            (unsigned)DXIL::StateObjectFlags::AllowStateObjectAdditions) != 0))
+        return true;
+    } break;
+    default:
+      break;
+    }
+  }
+  return false;
+}
+
 void DxilModule::CollectShaderFlagsForModule(ShaderFlags &Flags) {
-  for (Function &F : GetModule()->functions()) {
-    ShaderFlags funcFlags = ShaderFlags::CollectShaderFlags(&F, this);
-    Flags.CombineShaderFlags(funcFlags);
-  };
+  ComputeShaderCompatInfo();
+  for (auto &itInfo : m_FuncToShaderCompat)
+    Flags.CombineShaderFlags(itInfo.second.shaderFlags);
+
+  // Clear UsesDerivatives flag for module, making sure
+  // DerivativesInMeshAndAmpShaders is set for MS/AS.
+  if (Flags.GetUsesDerivatives()) {
+    Flags.SetUsesDerivatives(false);
+    if (m_pSM->IsMS() || m_pSM->IsAS())
```

Comment:
Why compute is not here?

Reply:
Advanced Texture Ops made derivatives in mesh and amplification shaders a separate feature bit. This is to check if that bit is needed.